python/tests/operations/reduce_over_samples.py

```diff
@@ -537,13 +539,402 @@ def test_reduction_allsamples(self):
 
         sum_X = fn.sum_over_samples(X, samples_names=["samples"])
         mean_X = fn.mean_over_samples(X, samples_names=["samples"])
+        var_X = fn.variance_over_samples(X, samples_names=["samples"])
+        std_X = fn.std_over_samples(X, samples_names=["samples"])
+
         self.assertTrue(fn.equal(sum_X, mean_X, only_metadata=True))
+        self.assertTrue(fn.equal(sum_X, std_X, only_metadata=True))
+        self.assertTrue(fn.equal(mean_X, var_X, only_metadata=True))
         self.assertTrue(sum_X[0].samples == Labels.single())
+        self.assertTrue(std_X[0].samples == Labels.single())
+
         self.assertTrue(np.all(sum_X[0].values == np.sum(X[0].values, axis=0)))
         self.assertTrue(np.all(mean_X[0].values == np.mean(X[0].values, axis=0)))
+        self.assertTrue(np.allclose(std_X[0].values, np.std(X[0].values, axis=0)))
+        self.assertTrue(np.allclose(var_X[0].values, np.var(X[0].values, axis=0)))
+
+
+class TestStdSamples(unittest.TestCase):
+    def test_std_samples_block(self):
+        tensor_se = equistore.io.load(
+            os.path.join(DATA_ROOT, "qm7-spherical-expansion.npz"),
+            use_numpy=True,
+        )
+        tensor_ps = equistore.io.load(
+            os.path.join(DATA_ROOT, "qm7-power-spectrum.npz"),
+            use_numpy=True,
+        )
+        # tensor_ps = fn.remove_gradients(tensor_ps)
+        tensor_se = fn.remove_gradients(tensor_se)
+
+        bl1 = tensor_ps[0]
+
+        reduce_tensor_se = fn.std_over_samples(tensor_se, samples_names="center")
+        reduce_tensor_ps = fn.std_over_samples(tensor_ps, samples_names=["center"])
+
+        self.assertTrue(
+            np.allclose(
+                np.std(bl1.values[:4], axis=0),
+                reduce_tensor_ps.block(0).values[0],
+                rtol=1e-13,
+            )
+        )
+
+        self.assertTrue(
+            np.all(
+                np.allclose(
+                    np.std(bl1.values[4:10], axis=0),
+                    reduce_tensor_ps.block(0).values[1],
+                    rtol=1e-13,
+                )
+            )
+        )
+        self.assertTrue(
+            np.allclose(
+                np.std(bl1.values[22:26], axis=0),
+                reduce_tensor_ps.block(0).values[5],
+                rtol=1e-13,
+            )
+        )
+        self.assertTrue(
+            np.allclose(
+                np.std(bl1.values[38:46], axis=0),
+                reduce_tensor_ps.block(0).values[8],
+                rtol=1e-13,
+            )
+        )
+        self.assertTrue(
+            np.all(
+                np.allclose(
+                    np.std(bl1.values[46:], axis=0),
+                    reduce_tensor_ps.block(0).values[9],
+                    rtol=1e-13,
+                )
+            )
+        )
```

Comment:
Also lets try to split this into individual tests in the future. This makes debugging in case of errors easier.